drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -634,6 +654,13 @@ void nrf_802154_receive_failed(nrf_802154_rx_error_t error)
 	nrf5_data.last_frame_ack_fpb = false;
 }
 
+void nrf_802154_tx_started(const uint8_t *frame)
```

Comment:
This function will be removed when we add support for nRF53. Bus we can move `k_sem_give()` to `transmitted_raw()` and to `transmit_failed()`.

Reply:
In such case the semaphore is pointless - it will duplicate the `nrf5_radio->tx_wait` semaphore.

Generally - good point, it would be great to check if there is a specific requirement on the `nrf5_tx_started(dev, pkt, frag);` callback in other protocols. This callback is the only reason for adding the new semaphore.